### PR TITLE
Test for GeoMapper

### PR DESCRIPTION
**Pull request by TomFischer**
PR contains a test for the class ` GeoMapper`. Test cases are generated using some new implemented autocheck generators (`IntervalGenerator` and `IntervalTupleGenerator`).

### CODE REVIEW

File: Tests/MeshGeoToolsLib/TestGeoMapper.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			"Test", MathLib::Point3d{ {{0.0, 0.0, 0.0}} },
+			MathLib::Point3d{ {{1.0, 1.0, 0.0}} }, {{110,60}},
+				[](double x, double y) { return std::cos(x+y); })};
+
+	ac::gtest_reporter gtest_reporter;
+};
+
+// map points on the surface mesh create by
+// MeshLib::MeshGenerator::createSurfaceMesh that approximates a given surface
+// function
```
Comment (endJunction) [inline Tests/MeshGeoToolsLib/TestGeoMapper.cpp]: Short comment about what property is tested, please.

File: Tests/MathLib/AutoCheckTools.h [hunk 2]
```diff
@@ -15,6 +15,58 @@
...
+};
+
+template <typename T, typename Gen = IntervalGenerator<T>>
+struct IntervalTupleGenerator
+{
+	IntervalTupleGenerator(Gen& ig0, Gen& ig1, Gen& ig2)
+	    : x_gen(ig0), y_gen(ig1), z_gen(ig2)
+	{}
+
+	Gen x_gen, y_gen, z_gen;
```
Comment (endJunction) [inline Tests/MathLib/AutoCheckTools.h]: Potentially dangerous.
Assume the `x_gen` is default generated, then the `IntervalGenerator::_m` and `_n` values are 0 and `IntervalGenerator::operator()` returns 0 always.
I think providing default values for `_m` and `_n` would be good solution. `_m = 1`, `_n = 0`.

**Comment by endJunction** [comment]
:fast_forward:

File: Tests/MathLib/AutoCheckTools.h (hunk 2 above)
Comment (TomFischer) [inline Tests/MathLib/AutoCheckTools.h]: Fixed.

File: Tests/MathLib/AutoCheckTools.h [hunk 3]
```diff
@@ -15,6 +15,58 @@
...
+	Gen generator;
+
+	using result_type = T;
+
+	result_type intervalMap(T val) const
+	{
+		return _m * val + _n;
+	}
+
+	result_type operator()(std::size_t size = 0)
```
Comment (chleh) [inline Tests/MathLib/AutoCheckTools.h]: The parameter `size` is not used, and that generates really ugly warning messages.
Please comment/remove that parameter. And do you need the default value? If not, the I would opt to remove it.

```
In file included from /home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/TestGeoMapper.cpp:19:0:
/home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/../MathLib/AutoCheckTools.h: In Instanziierung von »autocheck::IntervalTupleGenerator<T, Gen>::result_type autocheck::IntervalTupleGenerator<T, Gen>::operator()(std::size_t) [with T = double; Gen = autocheck::IntervalGenerator<double>; autocheck::IntervalTupleGenerator<T, Gen>::result_type = std::__debug::array<double, 3ul>; std::size_t = long unsigned int]«:
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:28:32:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t, const autocheck::range<0, Is ...>&) [with T = GeoLib::Point; Gens = {autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > >}; int ...Is = {0}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:33:34:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t) [with T = GeoLib::Point; Gens = {autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > >}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:286:48:   erfordert durch »autocheck::cons_generator<T, Gens>::result_type autocheck::cons_generator<T, Gens>::operator()(size_t) [with T = GeoLib::Point; Gens = {autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > >}; autocheck::cons_generator<T, Gens>::result_type = GeoLib::Point; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator_combinators.hpp:86:26:   erfordert durch »autocheck::detail::fixed_size_generator<Gen>::result_type autocheck::detail::fixed_size_generator<Gen>::operator()(size_t) [with Gen = autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > >; autocheck::detail::fixed_size_generator<Gen>::result_type = GeoLib::Point; size_t = long unsigned int]«
/usr/include/c++/5.3.0/bits/stl_algo.h:4337:18:   [ skipping 2 instantiation contexts, use -ftemplate-backtrace-limit=0 to disable ]
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator_combinators.hpp:86:26:   erfordert durch »autocheck::detail::fixed_size_generator<Gen>::result_type autocheck::detail::fixed_size_generator<Gen>::operator()(size_t) [with Gen = autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > >; autocheck::detail::fixed_size_generator<Gen>::result_type = std::__debug::vector<GeoLib::Point>; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:28:32:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t, const autocheck::range<0, Is ...>&) [with T = std::tuple<std::__debug::vector<GeoLib::Point, std::allocator<GeoLib::Point> > >; Gens = {autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > >}; int ...Is = {0}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/generator.hpp:33:34:   erfordert durch »T autocheck::generate(std::tuple<_Elements ...>&, size_t) [with T = std::tuple<std::__debug::vector<GeoLib::Point, std::allocator<GeoLib::Point> > >; Gens = {autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > >}; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/arbitrary.hpp:50:44:   erfordert durch »bool autocheck::arbitrary<Gens>::operator()(autocheck::value<std::tuple<typename autocheck::arbitrary<Gens>::ExpandArgs<Gens>::result_type ...> >&) [with Gens = {autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > >}]«
/home/lehmannc/prog/ogs6/github-chleh/src/ThirdParty/autocheck/include/autocheck/check.hpp:39:15:   erfordert durch »void autocheck::check(Property, size_t, Arbitrary, const autocheck::reporter&, autocheck::classifier<Args ...>, bool) [with Args = {std::__debug::vector<GeoLib::Point, std::allocator<GeoLib::Point> >}; Property = MeshGeoToolsLibGeoMapper_PointsOnSurfaceMesh_Test::TestBody()::<lambda(std::__debug::vector<GeoLib::Point>&)>; Arbitrary = autocheck::arbitrary<autocheck::detail::fixed_size_generator<autocheck::list_generator<autocheck::cons_generator<GeoLib::Point, autocheck::IntervalTupleGenerator<double, autocheck::IntervalGenerator<double, autocheck::generator<double, void> > > > > > >; size_t = long unsigned int]«
/home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/TestGeoMapper.cpp:87:17:   von hier erfordert
/home/lehmannc/prog/ogs6/github-chleh/src/Tests/MeshGeoToolsLib/../MathLib/AutoCheckTools.h:64:44: Warnung: unbenutzter Parameter »size« [-Wunused-parameter]
  result_type operator()(std::size_t size = 0)
```

**Pull request merged**